pkgs/development/python-modules/poetry/default.nix

```
@@ -60,6 +60,15 @@ in buildPythonPackage rec {
   ] ++ lib.optionals (isPy27 || isPy34) [ typing pathlib2 ]
     ++ lib.optionals isPy27 [ virtualenv functools32 ];
 
+  postInstall = ''
+    mkdir -p "$out/etc/bash_completion.d"
+    "$out/bin/poetry" completions bash > "$out/etc/bash_completion.d/poetry.bash-completion"
```

Use `/share/bash-completion/completions/` because is lazy loaded
```suggestion
    "$out/bin/poetry" completions bash > "$out/share/bash-completion/completions/poetry"
```

---

Thanks! Learning new things every day :)